include/dynamatic/Support/CFG.h

```diff
@@ -97,8 +97,26 @@ bool isBackedge(Value val, Operation *user, BBEndpoints *endpoints = nullptr);
 /// user (the function will assert if that is not the case).
 bool isBackedge(Value val, BBEndpoints *endpoints = nullptr);
 
-/// Determines whether the Handshake operation cannot belong to the implicit CFG
-/// of a Handshake function (i.e., cannot have a basic block attribute).
+/// Data structure to hold all edges belonging to a pair of endpoints. This
+/// constitutes an Arc. srcBB and dstBB can be equal when the arcs are
+/// Backedges. An edge is represented by means of an mlir::OpOperand.
+struct BBArc {
+  unsigned srcBB;
+  unsigned dstBB;
+  mlir::SmallVector<OpOperand *> edges;
+};
+
+/// Define a map from a BB's number to a vector of BBArcs. This can be used, for
+/// example, to map a BB to a set of arcs that lead to BBs that are predecessors
+/// in the CFG.
+using BBtoArcsMap = llvm::MapVector<unsigned, mlir::SmallVector<BBArc>>;
```

Comment:
The documentation above says "set of arcs" yet the value type is a vector (which I understand why, hashing the `edges` of a `BBArc` is not really possible) so it is slightly confusing.

Also, shouldn't the documentation end by "that _start with_ BBs that are predecessors in the CFG"? It seems weird to say that an arc lead to a predecessor. Alternatively, the example could be "that lead to BBs that are _successors_ in the CFG."

Reply:
The comment is improved with your suggestions. I agree on leaving this as a `mlir::SmallVector`.